src/Core/Component.js

```diff
@@ -53,19 +53,48 @@ function forceUpdate (callback) {
 	enqueueComponentUpdate(getComponentElement(this), this, callback, SharedComponentForceUpdate)
 }
 
+/**
+ * @param {object} descriptor
+ * @return {function}
+ */
+function createClass (description) {
+	function klass () {}
+
+	for (var name in description)
+		description[name] = {value: description[name]}
+
+	klass.prototype = create(Component[SharedSitePrototype], description)
+
+	return klass
+}
+
 /**
  * @param {Object} prototype
  * @return {Object}
  */
 function createComponentPrototype (prototype) {
-	prototype[SharedSiteForceUpdate] = forceUpdate
-	prototype[SharedSiteSetState] = setState
-	prototype[SharedSiteRender] = prototype[SharedSiteRender] || noop
-	prototype[SymbolComponent] = true
+	defineProperty(prototype, SymbolComponent, {value: SymbolComponent})
+	defineProperty(prototype, SharedSiteSetState, {value: setState})
+	defineProperty(prototype, SharedSiteForceUpdate, {value: forceUpdate})
+
+	if (!prototype[SharedSiteRender])
+		defineProperty(prototype, SharedSiteRender, {value: noop, configurable: true})
 
 	return prototype
 }
 
+/**
+ * @param {AsyncGenerator} generator
+ * @return {object}
+ */
+function createComponentGenerator (generator) {
+	return function then (resolve, reject) {
```

Comment:
shouldn't this be `new Promise function(resolve, reject) { }`

Reply:
The callsite where this is invoked looks like

```js
createElementPromise(createComponentGenerator(element))
```

Where `createElementPromise` looks like

```js
return createElement({then: callback})
```

Since we can work with "thenables" as well we don't need to rely on a the Promise constructor for this.

Reply:
And if it throws?